pkg/swagger/swagger.go

```diff
@@ -50,19 +55,31 @@ func Run(api, outputDir string) error {
 			Summary:     "Lists credentials of an OpenShift cluster with the specified subscription, resource group and resource name.",
 			Description: "The operation returns the credentials.",
 			OperationID: "OpenShiftClusters_ListCredentials",
-			Parameters:  populateParameters(3, "OpenShiftCluster", "OpenShift cluster"),
-			Responses:   populateResponses("OpenShiftClusterCredentials", false, http.StatusOK),
+			Parameters:  g.populateParameters(3, "OpenShiftCluster", "OpenShift cluster"),
+			Responses:   g.populateResponses("OpenShiftClusterCredentials", false, http.StatusOK),
 		},
 	}
+	if g.kubeconfig {
+		s.Paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenShift/openShiftClusters/{resourceName}/listAdminCredentials"] = &PathItem{
+			Post: &Operation{
+				Tags:        []string{"OpenShiftClusters"},
+				Summary:     "Lists admin kubeconfig of an OpenShift cluster with the specified subscription, resource group and resource name.",
+				Description: "The operation returns the admin kubeconfig.",
+				OperationID: "OpenShiftClusters_ListAdminKubeconfig",
+				Parameters:  g.populateParameters(3, "OpenShiftCluster", "OpenShift cluster"),
+				Responses:   g.populateResponses("OpenShiftClusterAdminKubeconfig", false, http.StatusOK),
+			},
+		}
+	}
 
 	s.Paths["/providers/Microsoft.RedHatOpenShift/operations"] = &PathItem{
```

Comment:
I think we need to move `listCredentials`, `listAdminCredentials` (if we actually need it, see the question above) and `operations` paths somewhere into `generator`. Similarly to `populateTopLevelPaths`.

And probably all logic from `Run` into generator so the whole `Run` function will be something like:

```go
g, err := New(api, params1, param2)
if err != nil {
	return err
}

s := g.Swagger()

b, err := json.MarshalIndent(s, "", "  ")
if err != nil {
	return err
}

b = append(b, '\n')

return ioutil.WriteFile(outputDir+"/redhatopenshift.json", b, 0666)
```

 But I think it should be a separate PR: this one is already too big.